pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -568,75 +573,48 @@ func (p *Provisioner) populateDomainAndPort(sc *storagev1.StorageClass) error {
-	var quotaEnabled bool
-
-	maxObjects := MaxObjectQuota(options.ObjectBucketClaim.Spec.AdditionalConfig)
-	if maxObjects != "" {
-		quotaEnabled = true
-
-		maxObjectsInt64, err = toInt64(maxObjects)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse maxObjects quota for user %q", p.cephUserName)
-		}
-	}
-
-	maxSize := MaxSizeQuota(options.ObjectBucketClaim.Spec.AdditionalConfig)
-	if maxSize != "" {
-		quotaEnabled = true
-
-		maxSizeInt64, err = toInt64(maxSize)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse maxSize quota for user %q", p.cephUserName)
-		}
+	additionalConfig, err := additionalConfigSpecFromMap(options.ObjectBucketClaim.Spec.AdditionalConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to process additionalConfig")
 	}
 
 	objectUser, err := p.adminOpsClient.GetUser(p.clusterInfo.Context, admin.User{ID: p.cephUserName})
 	if err != nil {
 		return errors.Wrapf(err, "failed to fetch user %q", p.cephUserName)
 	}
+	currentQuota := objectUser.UserQuota
+	// copy QuotaSpec as a means of ignoring fields that are not being managed
+	targetQuota := objectUser.UserQuota
 
 	// enable or disable quota for user
-	if *objectUser.UserQuota.Enabled != quotaEnabled {
-		err = p.adminOpsClient.SetUserQuota(p.clusterInfo.Context, admin.QuotaSpec{UID: p.cephUserName, Enabled: &quotaEnabled})
-		if err != nil {
-			return errors.Wrapf(err, "failed to set user %q quota enabled=%v for obc", p.cephUserName, quotaEnabled)
-		}
-	}
+	quotaEnabled := (additionalConfig.maxObjects != nil) || (additionalConfig.maxSize != nil)
 
-	if !quotaEnabled {
-		// no need to process anything else if quotas are disabled
-		return nil
+	targetQuota.Enabled = &quotaEnabled
+	if additionalConfig.maxObjects == nil {
+		var objects int64 = -1
+		targetQuota.MaxObjects = &objects
+	} else {
+		targetQuota.MaxObjects = additionalConfig.maxObjects
 	}
-
-	if *objectUser.UserQuota.MaxObjects != maxObjectsInt64 {
-		err = p.adminOpsClient.SetUserQuota(p.clusterInfo.Context, admin.QuotaSpec{UID: p.cephUserName, MaxObjects: &maxObjectsInt64})
-		if err != nil {
-			return errors.Wrapf(err, "failed to set MaxObjects=%v to user %q", maxObjectsInt64, p.cephUserName)
-		}
+	if additionalConfig.maxSize == nil {
+		var size int64 = -1
+		targetQuota.MaxSize = &size
+	} else {
+		targetQuota.MaxSize = additionalConfig.maxSize
 	}
 
-	if objectUser.UserQuota.MaxSize != &maxSizeInt64 {
-		err = p.adminOpsClient.SetUserQuota(p.clusterInfo.Context, admin.QuotaSpec{UID: p.cephUserName, MaxSize: &maxSizeInt64})
+	if !reflect.DeepEqual(currentQuota, targetQuota) {
+		// UID is not set in the QuotaSpec returned by GetUser()
+		targetQuota.UID = p.cephUserName
+		err = p.adminOpsClient.SetUserQuota(p.clusterInfo.Context, targetQuota)
 		if err != nil {
-			return errors.Wrapf(err, "failed to set MaxSize=%v to user %q", maxSizeInt64, p.cephUserName)
+			return errors.Wrapf(err, "failed to set user %q quota enabled=%v", p.cephUserName, quotaEnabled)
```

Comment:
It might be helpful to print the desired/controlled part of the spec too, so we can verify Rook is interpreting what is given in the way we expect.
 
```suggestion
			return errors.Wrapf(err, "failed to set user %q quota enabled=%v %+v", p.cephUserName, quotaEnabled, additionalConfigSpec)
```